core/iwasm/interpreter/wasm_loader.c

```diff
@@ -4054,17 +4062,47 @@ load_global_section(const uint8 *buf, const uint8 *buf_end, WASMModule *module,
     return false;
 }
 
+static bool
+check_duplicate_exports(WASMModule *module, char *error_buf,
+                        uint32 error_buf_size)
+{
+    char **names = loader_malloc(module->export_count * sizeof(char *),
+                                 error_buf, error_buf_size);
+    bool result = false;
+
+    if (!names) {
+        return result;
+    }
+
+    for (uint32 i = 0; i < module->export_count; i++) {
+        names[i] = module->exports[i].name;
+    }
+
+    qsort(names, module->export_count, sizeof(char *), cmp_export_name);
+
+    for (uint32 i = 1; i < module->export_count; i++) {
+        if (!strcmp(names[i], names[i - 1])) {
+            set_error_buf(error_buf, error_buf_size, "duplicate export name");
+            goto cleanup;
+        }
+    }
```

Comment:
This change will allocate memory and qsort the array each time, not sure whether it improves the performance a lot? We ever discussed it before, and I think a better way is to create the name to index map array (element index in the exports array) in the wasm module, and use the binary search algorithm to lookup export with the name: get the element index in the exports array, use it to get the export and then use the `export->index` to get the function instance in wasm instance.

And here, seems we can establish the map array with binary search algorithm also: get the right place the insert the new export, and then insert it.

ps. https://github.com/bytecodealliance/wasm-micro-runtime/issues/3407.